app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/notification/AppTPReminderNotificationScheduler.kt

```diff
@@ -57,12 +59,20 @@ class DeviceShieldReminderNotificationScheduler @Inject constructor(
     @AppCoroutineScope private val appCoroutineScope: CoroutineScope,
     private val vpnReminderNotificationBuilder: VpnReminderNotificationBuilder,
     private val vpnReminderNotificationContentPluginPoint: PluginPoint<VpnReminderNotificationContentPlugin>,
+    private val appTrackingProtection: AppTrackingProtection,
 ) : VpnServiceCallbacks {
+    private var isAppTPEnabled: AtomicReference<Boolean> = AtomicReference(false)
 
     override fun onVpnStarted(coroutineScope: CoroutineScope) {
-        scheduleUndesiredStopReminderAlarm()
-        cancelReminderForTomorrow()
-        hideReminderNotification()
+        runBlocking {
+            isAppTPEnabled.set(appTrackingProtection.isEnabled())
+        }
+        if (isAppTPEnabled.get()) {
+            // These are all relevant for when AppTP has been enabled.
+            scheduleUndesiredStopReminderAlarm()
+            cancelReminderForTomorrow()
+            hideReminderNotification()
+        }
         enableReminderReceiver()
```

Comment:
Why the runblocking? can't we do this instead?
```suggestion
        coroutineScope.launch {
            isAppTPEnabled.set(appTrackingProtection.isEnabled())
            if (isAppTPEnabled.get()) {
                // These are all relevant for when AppTP has been enabled.
                scheduleUndesiredStopReminderAlarm()
                cancelReminderForTomorrow()
                hideReminderNotification()
            }
            enableReminderReceiver()
        }
```